Grammar/python.gram

```diff
@@ -978,6 +979,16 @@ invalid_case_block:
 invalid_as_pattern:
     | or_pattern 'as' a="_" { RAISE_SYNTAX_ERROR_KNOWN_LOCATION(a, "cannot use '_' as a target") }
     | or_pattern 'as' !NAME a=expression { RAISE_SYNTAX_ERROR_KNOWN_LOCATION(a, "invalid pattern target") }
+invalid_class_pattern:
+    | name_or_attr '(' a=invalid_class_argument_pattern ','? ')' { RAISE_SYNTAX_ERROR_KNOWN_RANGE(
+        PyPegen_first_item(a, pattern_ty),
+        PyPegen_last_item(a, pattern_ty),
+        "positional patterns follow keyword patterns") }
+invalid_class_argument_pattern[asdl_pattern_seq*]:
+    | positional_patterns ',' keyword_patterns ',' a=positional_patterns { a }
+    | keyword_patterns ',' a=positional_patterns ',' keyword_patterns { a }
+    | keyword_patterns ',' a=positional_patterns { a }
```

Comment:
Hm. Would something like this (untested) catch _all_ invalid cases where a positional pattern follows a keyword one?

```suggestion
invalid_class_argument_pattern[asdl_pattern_seq*]:
    | [positional_patterns ','] keyword_patterns ',' a=positional_patterns [',' ','.(keyword_patterns | positional_patterns)+] { a }
```

Reply:
I'm imagining a pathological case like `C(a=b, c, d=e, f, g=h, i, j=k)`, for example. Maybe it's not worth catching, but it seems like it would be easy enough to do.

Reply:
I think we can cover that if we rely on eager parsing on the rule above:

```
>>> match x:
...    case C(a=b, c, d=e, f, g=h, i, j=k, ...):
  File "<stdin>", line 2
    case C(a=b, c, d=e, f, g=h, i, j=k, ...):
                ^
SyntaxError: positional patterns follow keyword patterns
```